src/domain/create-entry.ts

```diff
@@ -136,34 +159,42 @@ export class CreateEntryService {
     await this.gitClient.push(bcr.diskPath, "authed-fork", branch);
   }
 
-  private async stampSourceFile(
-    sourcePath: string,
-    destPath: string,
-    rulesetRepo: Repository,
+  // The version in the archived MODULE.bazel version should match the release version.
+  // If it doesn't, add a patch to set the correct version. This is useful when a release
+  // archive is just an archive of the source, and the source MODULE.bazel is kept unstamped
+  // (e.g., has '0.0.0' as the version).
+  private patchModuleVersionIfMismatch(
+    moduleFile: ModuleFile,
     version: string,
-    tag: string
-  ): Promise<void> {
-    // Substitute variables into source.json
-    const sourceContent = fs.readFileSync(sourcePath, { encoding: "utf-8" });
-    const substituted = sourceContent
-      .replace(/{REPO}/g, rulesetRepo.name)
-      .replace(/{OWNER}/g, rulesetRepo.owner)
-      .replace(/{VERSION}/g, version)
-      .replace(/{TAG}/g, tag);
-
-    // Compute the integrity hash
-    const sourceJson = JSON.parse(substituted);
-
-    const digest = await this.releaseHashService.calculate(sourceJson.url);
-    sourceJson.integrity = `sha256-${digest}`;
-
-    fs.writeFileSync(
-      destPath,
-      `${JSON.stringify(sourceJson, undefined, 4)}\n`,
-      {
-        encoding: "utf-8",
-      }
-    );
+    sourceTemplate: SourceTemplate,
+    bcrVersionEntryPath: string
+  ): void {
+    if (moduleFile.version !== version) {
```

Comment:
Hmm, I wonder if we ought to error if the moduleFile.version is just wrong, as opposed to clearly unstamped. Maybe a releaser would want to know that they forgot a bumpversion commit and their module is off by one.
However, other than assuming everyone likes our `0.0.0` pattern, I'm not sure how we could tell which ones are intentional.

Reply:
I considered this, but I wasn't sure we should assume that people use `0.0.0` to mean unstamped. If there was an unintentional version in the release archive's MODULE, at least it'll show in the PR patch. I'm not sure it's the app's responsibility to catch this.